### PR TITLE
FluidBuild: support project references

### PR DESCRIPTION
**Pull request by curtisman**
In preparation to split test *.spec.ts files into a separate directory from the src for each packages.  Project reference will be used so that test files can reference the main source files but can have different config and separate build invocation for them.

### CODE REVIEW

**Comment by vladsud** [comment]
I'm curious about direction. I.e. is this required to split test code out of packages? It will be harder to keep things in sync.

File: tools/build-tools/src/fluidBuild/fluidPackageCheck.ts [hunk 1]
```diff
@@ -223,7 +224,13 @@ export class FluidPackageCheck {
 
             const buildPrefix = pkg.getScript("build:genver") ? "npm run build:genver && " : "";
             if (pkg.getScript("tsc")) {
-                buildCompile.push("tsc");
+                if (pkg.getScript("build:test")) {
+                    buildCommonJs.push("tsc");
+                    buildCommonJs.push("build:test");
+                    buildCompile.push("build:commonjs");
```
Comment (vladsud) [inline tools/build-tools/src/fluidBuild/fluidPackageCheck.ts:230]: Curious, why is this needed?

**Review by vladsud — APPROVED** [review]

File: tools/build-tools/src/fluidBuild/fluidPackageCheck.ts (hunk 1 above)
Comment (curtisman) [reply tools/build-tools/src/fluidBuild/fluidPackageCheck.ts:230]: This is to prepare for having a new script in the package.json `build:test` for building the tests separate from the source.  This is here to enforce some uniformity of the packages if they have the build:test script.

**Comment by curtisman** [comment]
> I'm curious about direction. I.e. is this required to split test code out of packages? It will be harder to keep things in sync.

It will be the same package, but in separate directory called `test`, so that they can be built using different config then the source.  This can avoid introducing the `node` or `mocha` typing the config to build source.

**Comment by vladsud** [comment]
Got it - thanks for explanation.
